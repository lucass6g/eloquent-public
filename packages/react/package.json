{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
