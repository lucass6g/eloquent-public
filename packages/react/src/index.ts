export {Switch, Case, Default} from './components/conditional-switch'