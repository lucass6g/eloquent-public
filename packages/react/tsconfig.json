{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
