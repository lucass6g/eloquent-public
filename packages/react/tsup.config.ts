import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})