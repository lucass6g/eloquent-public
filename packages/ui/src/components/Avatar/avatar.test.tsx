import { it, describe, expect, beforeAll, afterAll, beforeEach } from "vitest"
import { render, RenderResult, screen, waitFor } from '@testing-library/react';
import { Avatar } from '.';


describe('Avatar Component', () => {
    it('renders Avatar with fallback', () => {
        const component = render(
            <Avatar.Root>
                <Avatar.Fallback>Elo</Avatar.Fallback>
            </Avatar.Root>
        );
        const fallback = screen.getByText('Elo');
        expect(fallback).toBeInTheDocument();
        expect(component.baseElement).toMatchSnapshot();
    });
    it('applies correct size variants classes', async () => {
        const component = render(
            <Avatar.Root data-testid="root-element" size="xl">
                <Avatar.Fallback fontSize="lg">Elo</Avatar.Fallback>
            </Avatar.Root>
        );
        let rootElement = screen.getByTestId('root-element');
        let fallbackElement = screen.getByText('Elo');

        await waitFor(() => {
            expect(rootElement).toHaveClass('h-24 w-24');
            expect(fallbackElement).toHaveClass('text-lg');
            expect(component.baseElement).toMatchSnapshot();
        })
        component.rerender(
            <Avatar.Root size="sm">
                <Avatar.Fallback fontSize="sm">Elo</Avatar.Fallback>
            </Avatar.Root>
        );

        await waitFor(() => {
            fallbackElement = screen.getByText('Elo');
            expect(rootElement).toHaveClass('h-10 w-10');
            expect(fallbackElement).toHaveClass('text-sm');
            expect(component.baseElement).toMatchSnapshot();
        })
    });
});

describe('Avatar with valid image', () => {
    let rendered: RenderResult;
    let image: HTMLElement | null = null;
    const orignalGlobalImage = window.Image;
  
    beforeAll(() => {
      (window.Image as any) = class MockImage {
        onload: () => void = () => {};
        src: string = '';
        constructor() {
          setTimeout(() => {
            this.onload();
          }, 300);
          return this;
        }
      };
    });
  
    afterAll(() => {
      window.Image = orignalGlobalImage;
    });
  
    beforeEach(() => {
      rendered = render(
        <Avatar.Root data-testid='avatar-root'>
          <Avatar.Fallback>Avatar Shad</Avatar.Fallback>
          <Avatar.Image src="/test" alt="@shadcn" />
        </Avatar.Root>
      );
    });
  
    it('should render the fallback initially', () => {
      const fallback = rendered.queryByText('Avatar Shad');
      expect(fallback).toBeInTheDocument();
      expect(rendered.baseElement).toMatchSnapshot();
    });
  
    it('should render the image after it has loaded', async () => {
      image = await rendered.findByRole('img');
      expect(image).toBeInTheDocument();
      expect(rendered.baseElement).toMatchSnapshot();
    });
  
    it('should have alt text on the image', async () => {
      image = await rendered.findByAltText('@shadcn');
      expect(image).toBeInTheDocument();
      expect(rendered.baseElement).toMatchSnapshot();
    });
  });