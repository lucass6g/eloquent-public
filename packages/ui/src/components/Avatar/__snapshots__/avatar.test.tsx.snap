// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar Component > applies correct size variants classes 1`] = `
<body>
  <div>
    <span
      class="relative flex shrink-0 overflow-hidden rounded-full h-24 w-24"
      data-testid="root-element"
    >
      <span
        class="flex h-full w-full items-center justify-center rounded-full bg-muted text-lg"
      >
        Elo
      </span>
    </span>
  </div>
</body>
`;

exports[`Avatar Component > applies correct size variants classes 2`] = `
<body>
  <div>
    <span
      class="relative flex shrink-0 overflow-hidden rounded-full h-10 w-10"
    >
      <span
        class="flex h-full w-full items-center justify-center rounded-full bg-muted text-sm"
      >
        Elo
      </span>
    </span>
  </div>
</body>
`;

exports[`Avatar Component > applies correct size variants classes 3`] = `
<body>
  <div>
    <span
      class="relative flex shrink-0 overflow-hidden rounded-full h-24 w-24"
      data-testid="root-element"
    >
      <span
        class="flex h-full w-full items-center justify-center rounded-full bg-muted text-lg"
      >
        Elo
      </span>
    </span>
  </div>
</body>
`;

exports[`Avatar Component > applies correct size variants classes 4`] = `
<body>
  <div>
    <span
      class="relative flex shrink-0 overflow-hidden rounded-full h-10 w-10"
    >
      <span
        class="flex h-full w-full items-center justify-center rounded-full bg-muted text-sm"
      >
        Elo
      </span>
    </span>
  </div>
</body>
`;

exports[`Avatar Component > renders Avatar with fallback 1`] = `
<body>
  <div>
    <span
      class="relative flex shrink-0 overflow-hidden rounded-full h-16 w-16"
    >
      <span
        class="flex h-full w-full items-center justify-center rounded-full bg-muted text-base"
      >
        Elo
      </span>
    </span>
  </div>
</body>
`;

exports[`Avatar Component > renders Avatar with image 1`] = `
<body>
  <div>
    <span
      class="relative flex shrink-0 overflow-hidden rounded-full h-16 w-16"
    />
  </div>
</body>
`;

exports[`Avatar with valid image > should have alt text on the image 1`] = `
<body>
  <div>
    <span
      class="relative flex shrink-0 overflow-hidden rounded-full h-16 w-16"
      data-testid="avatar-root"
    >
      <img
        alt="@shadcn"
        class="aspect-square h-full w-full"
        src="/test"
      />
    </span>
  </div>
</body>
`;

exports[`Avatar with valid image > should render the fallback initially 1`] = `
<body>
  <div>
    <span
      class="relative flex shrink-0 overflow-hidden rounded-full h-16 w-16"
      data-testid="avatar-root"
    >
      <span
        class="flex h-full w-full items-center justify-center rounded-full bg-muted text-base"
      >
        Avatar Shad
      </span>
    </span>
  </div>
</body>
`;

exports[`Avatar with valid image > should render the image after it has loaded 1`] = `
<body>
  <div>
    <span
      class="relative flex shrink-0 overflow-hidden rounded-full h-16 w-16"
      data-testid="avatar-root"
    >
      <img
        alt="@shadcn"
        class="aspect-square h-full w-full"
        src="/test"
      />
    </span>
  </div>
</body>
`;
